rfc/0052-sdk3-full-text-search.md

```diff
@@ -566,7 +566,275 @@ A query that matches nothing.
+  * The `k` parameter is mandatory.  The FTS team suggest 3 is a sensible default to send from the SDKs.
+* There are new FTS index definition fields for defining the vector index, but they are inside the `plan` top-level field.  
+So there is nothing extra required from the SDKs to support these, as we already simply turn the untyped `plan` parameters into JSON.
+* The SDK (and Couchbase in general), at least in this first phase, have nothing to do with generating the vectors - both those stored in the documents, and those used for queries.
+The user will be responsible for generating these, for example using the OpenAI Embeddings API.
+* The feature works with both global and scoped FTS indexes.  As this is a 7.6+ feature, we will make scoped indexes the primary target for documentation, examples, etc.
+
+SDK PM advises that the most common vector search use-cases are likely to be, in this order:
+
+* Performing a vector search in isolation (without a traditional FTS query).
+* A SQL++ statement is used to also perform a vector search.
+  * This is just passed in the string statement and is expected to return the same query results JSON as at present, so from the SDK POV it's just a passthrough with no SDK modifications needed.
+* A vector search combined with a traditional FTS query.
+
+## API
+
+The FTS JSON request has fundamentally changed, as it now allows traditional FTS queries and/or vector queries.
+In addition, we need to consider that it may allow other query types in the future.
+
+Such a fundamental conceptual change necessitates a similarly fundamental change at the SDK level.
+
+Let us name the concepts we are working with:
+
+* The top-level search request.  This is currently an unnamed concept in the SDKs - let us term it a `SearchRequest`.
+* The traditional FTS query, which is already named `SearchQuery` in the SDKs.
+Includes the query itself and any options that apply only to the FTS query such as "boost".
+It's mapped to the top-level `query` JSON field.
+* An individual vector query, which we can call a `VectorQuery`.  These map to the members of the new top-level `knn` array.
+It can contain options applicable just to that vector query such as "boost" and "k".
+* A way to perform multiple `VectorQuery`s, which we can call a `VectorSearch`.
+Maps to the top-level `knn` array.
+* Any options applicable to all forms of query (both the `SearchQuery` and the `VectorSearch`), which are in `SearchOptions` (skip, limit, explain, highlight, fields, etc.)  We can continue to use the existing `SearchOptions` object for this.
+* Any options applicable only to the `VectorSearch`, which we can call `VectorSearchOptions`.
+
+So the user will be performing a `SearchRequest` containing a `SearchQuery` and/or a `VectorSearch` and/or a `FutureSearchType`, optionally with some `SearchOptions` applying to all of those.  
+A `VectorSearch` comprises 1+ `VectorQuery`s and optionally some `VectorSearchOptions`.
```

Comment:
The distinction between "search" and "query" is pretty arbitrary here - there's quite a few concepts to name and limited ways of doing so.  Open to alternative naming schemes!